pages/implementation-status.md

```diff
@@ -0,0 +1,202 @@
+1. Swift has had time to empirically demonstrate the success of its approach to generic programming; and
+2. Val and Swift share very similar type systems, so we expect the concepts developed in the latter to be portable in the former.
+
+### Write and publish the specification
+
+A complete specification of the language will increase confidence in the soundness of the language design.
+It will also serve as a reference document to write the implementation.
+
+The specification is currently an [ongoing work](https://github.com/val-lang/specification) that we expect to be delivered together with an alpha version of the language implementation.
+However, sections that relate to the semantics of the language still require significant effort.
+
+We also plan on publishing the specification in the form of a website or [GitBook](https://www.gitbook.com).
+
+## 2023
+
+Our main objectives for 2023 will be to
+- develop the missing features of the language;
+- investigate optimal implementation strategies implement an efficient compiler;
+- implement two-way interoperability with C++; and
+- develop an inclusive community.
+
+### Implement guaranteed optimizations
+
+Val is designed in such a way that it can communicate runtime costs very transparently.
+As a result, the language can offer several guaranteed optimization.
+For example, Val guarantees that closures do not require heap allocation unless the type of their environment is erased.
+We plan on implementing these guaranteed optimizations over Q1 and Q2.
+
+### Investigate implementation strategies for concurrency
+
+Once we will have finalized the design of Val's concurrency model, we will be able to investigate the optimal implementation strategies.
+
+We are leaning toward a [stackful coroutines](https://dl.acm.org/doi/10.1145/1462166.1462167), mostly to avoid monadic asynchronous programming (a.k.a. [polychromatic functions](https://journal.stuffwithstuff.com/2015/02/01/what-color-is-your-function/)), where asynchronous operations can only be called from asynchronous contexts.
+Nonetheless, discussions with experts have revealed that such a strategy may induce unfortunate runtime costs and that stackless coroutines may not necessarily force a monadic asynchronous style upon the language.
+
+One solution might be to adopt a sackful model as a default strategy and surface stack information in the type system to let the compiler opt for stackless code generation whenever possible.
+
+We on plan having adopted and implemented a strategy by the end of Q3.
+
+### Implement two-way interoperability with C++
+
+The objective of the second phase of C++ interoperability is to support seamless interaction of Val and C++ code in the same project.
+To that end, we will interface with clang to let Val compiler understand C++ and generate appropriate wrappers.
```

Comment:
```suggestion
To that end, we will interface with clang to let Val compiler understand C++ and generate any necessary glue code.
```

“wrapper” is a bit of a hot-button word, since Herb has said that a wrapper approach to interop doesn't work.